pyomo/contrib/doe/examples/fim_doe_tutorial.ipynb

```diff
@@ -57,77 +137,341 @@
+    "    assert (\n",
+    "        control_time[0] >= t_range[0] and control_time[-1] <= t_range[1]\n",
+    "    ), \"control time is outside time range.\"\n",
+    "\n",
+    "    if model_option == ModelOptionLib.stage1:\n",
+    "        mod.T = pyo.Var(\n",
+    "            mod.t_con,\n",
+    "            initialize=controls,\n",
+    "            bounds=(300, 700),\n",
+    "            within=pyo.NonNegativeReals,\n",
+    "        )\n",
+    "        return\n",
+    "\n",
+    "    else:\n",
+    "        para_list = [\"A1\", \"A2\", \"E1\", \"E2\"]\n",
+    "\n",
+    "        # Add variables\n",
+    "        mod.CA_init = CA_init\n",
+    "        mod.para_list = para_list\n",
+    "\n",
+    "        # Timepoints\n",
+    "        mod.t = ContinuousSet(bounds=t_range, initialize=control_time)\n",
+    "\n",
+    "        # Time-dependent design variable, initialized with the first control value\n",
+    "        def T_initial(m, t):\n",
+    "            if t in m.t_con:\n",
+    "                return controls[t]\n",
+    "            else:\n",
+    "                # Count how many control points are before the current t;\n",
+    "                # Locate the nearest neighbouring control point before this t\n",
+    "                neighbour_t = max(tc for tc in control_time if tc < t)\n",
+    "                return controls[neighbour_t]\n",
+    "\n",
+    "        mod.T = pyo.Var(\n",
+    "            mod.t, initialize=T_initial, bounds=(300, 700), within=pyo.NonNegativeReals\n",
+    "        )\n",
+    "\n",
+    "        mod.R = 8.31446261815324  # J / K / mole\n",
+    "\n",
+    "        # Define parameters as Param\n",
+    "        mod.A1 = pyo.Var(initialize=theta[\"A1\"])\n",
+    "        mod.A2 = pyo.Var(initialize=theta[\"A2\"])\n",
+    "        mod.E1 = pyo.Var(initialize=theta[\"E1\"])\n",
+    "        mod.E2 = pyo.Var(initialize=theta[\"E2\"])\n",
+    "\n",
+    "        # Concentration variables under perturbation\n",
+    "        mod.C_set = pyo.Set(initialize=[\"CA\", \"CB\", \"CC\"])\n",
+    "        mod.C = pyo.Var(\n",
+    "            mod.C_set, mod.t, initialize=C_init, within=pyo.NonNegativeReals\n",
+    "        )\n",
+    "\n",
+    "        # Time derivative of C\n",
+    "        mod.dCdt = DerivativeVar(mod.C, wrt=mod.t)\n",
+    "\n",
+    "        # Kinetic parameters\n",
+    "        def kp1_init(m, t):\n",
+    "            return m.A1 * pyo.exp(-m.E1 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        def kp2_init(m, t):\n",
+    "            return m.A2 * pyo.exp(-m.E2 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        mod.kp1 = pyo.Var(mod.t, initialize=kp1_init)\n",
+    "        mod.kp2 = pyo.Var(mod.t, initialize=kp2_init)\n",
+    "\n",
+    "        def T_control(m, t):\n",
+    "            \"\"\"\n",
+    "            T at interval timepoint equal to the T of the control time point at the beginning of this interval\n",
```

Comment:
Just a comment: This docstring is kind of confusing. I had to read it several times, and I'm still not sure I 100% know what it means. Maybe try to rephrase/simplify?